state/factory/factory.go

```diff
@@ -220,7 +227,22 @@ func (sf *factory) Start(ctx context.Context) error {
 	default:
 		return err
 	}
-	return sf.lifecycle.OnStart(ctx)
+	sf.mutex.Unlock()
+	needUnlock = false
+
+	// start the protocols
+	ctx = protocol.WithStateReaderCtx(ctx, protocol.StateReaderCtx{SR: sf})
+	if bcCtx, ok := protocol.GetBlockchainCtx(ctx); ok {
+		for _, p := range bcCtx.Registry.All() {
+			if s, ok := p.(lifecycle.Starter); ok {
+				if err := s.Start(ctx); err != nil {
+					return errors.Wrap(err, "failed to start protocol")
+				}
+			}
+		}
+	}
+	// push private data from protocols
+	return sf.dock.Push()
```

Comment:
protocol.Start() may read data from DB and generates their initial view, so call Push() here to commit the initial view